server/backends/blobstore/aws/aws.go

```diff
@@ -254,3 +257,53 @@ func (a *AwsS3BlobStore) BlobExists(ctx context.Context, blobName string) (bool,
 
 	return true, nil
 }
+
+func (a *AwsS3BlobStore) Writer(ctx context.Context, blobName string) (interfaces.CommittedWriteCloser, error) {
+	// Open a pipe.
+	pr, pw := io.Pipe()
+
+	errch := make(chan error, 1)
+
+	ctx, cancel := context.WithCancel(ctx)
+
+	// Upload from pr in a separate Go routine.
+	go func() {
+		defer pr.Close()
+		_, err := a.uploader.UploadWithContext(
+			ctx,
+			&s3manager.UploadInput{
+				Bucket: a.bucket,
+				Key:    aws.String(util.BlobPath(blobName)),
+				Body:   pr,
+			},
+		)
+		errch <- err
+		close(errch)
+	}()
+
+	zw := util.NewCompressWriter(pw)
+	cwc := ioutil.NewCustomCommitWriteCloser(zw)
+	cwc.CommitFn = func(int64) error {
+		if compresserCloseErr := zw.Close(); compresserCloseErr != nil {
+			cancel() // Don't try to finish the commit op if Close() failed.
+			if pipeCloseErr := pw.Close(); pipeCloseErr != nil {
+				log.Errorf("Error closing the pipe for %s: %s", blobName, pipeCloseErr)
+			}
+			<-errch // Canceling the context makes any error here meaningless.
```

Comment:
nit: this channel receive can be deleted since the channel is buffered. it'll just collect the write error once line 280 is hit, then get garbage collected.

Reply:
Thanks, didn't know; done.